### ISSUE TITLE
Use pageId instead of Spacekey and parent title 

### ISSUE THREAD

**Issue by muchino** [bsorrentino/maven-confluence-plugin#107]
Hi

At all let me say Thank you for this maven plugin I just added this plugin to our maven Archetypes to generate our documentation in the future in a better way.  

While I tested the plugin there came up the question in my mind "what should I do if the parent page will be renamed"  ok editing the pom again. 

But wouldn't it be nice to just use the pageId (maybe in a special "usepageid=true"  mode. 

So we can save us to define the Spacekey and the page title which can contain special characaters. 

What do you think about this? 

Cheers 
Oli

**Comment by bsorrentino** [comment]
Hi, thx for interest

It's a good technical enhancement ...  my opinion is that i would like preserve readability of  plugin configuration itself and ... using ID probably it will not 

However i think that it could be implemented in 2 ways:

1. Adding new parameter ```parentPageId```
1. As fallback strategy of ```parentPageTitle``` (if doesn't find title assume that it is an ID)

I'll add such feature as soon as possible (_4.10_)

**Comment by muchino** [comment]
Sound great :) I think the new parameter would be the best solution if you like to persist both name of the parameters. 

So you can define in your documentation: "If you configure the parentPageTitle, you need to specify the spaceKey. If you use the parentPageId, the spaceKey isn't needed anymore". 

And Because of the "Title" in the name of the parameter parentPageTitle, it can be confusing that the pageId is working there too.

**Comment by bsorrentino** [comment]
agree 

however, in the case of providing  ```parentPageId``` i get the parent page title in order to use it as prefix (unless the  ```childrenTitlesPrefixed``` is false)

**Comment by muchino** [comment]
sry - now I can't follow you :(

**Comment by bsorrentino** [comment]
I've deployed release ```4.10-SNAPSHOT``` with fix.
Now is available the argument ```parentPageId```, please test it and le me know

**Comment by muchino** [comment]
what should I need to to to use the snapshot? where is the repository

**Comment by bsorrentino** [comment]
the snapshot repo is [here](https://oss.sonatype.org/content/repositories/snapshots)

to configure in maven 

```xml
<pluginRepositories>

    <!-- IF YOU WANT STAY TUNED ON UPDATE REMOVE COMMENT -->
    <pluginRepository>
        <id>sonatype-repo</id>
        <url>https://oss.sonatype.org/content/repositories/snapshots</url>
        <releases>
            <enabled>false</enabled>
        </releases>
        <snapshots>
            <enabled>true</enabled>
        </snapshots>
    </pluginRepository>

</pluginRepositories>
```

**Comment by muchino** [comment]
works - cool :) 

what is now the order if both is added? 
If pageID is present - this one is used ?

**Comment by bsorrentino** [comment]
Hi, Thanks for feedback

If both are present pageID has priority (beeing more efficient)

I'll arrange official release soon

**Comment by bsorrentino** [comment]
I've released the release `4.10`

**Comment by muchino** [comment]
Thank you!
Am 01.07.2016 12:15 schrieb "bsorrentino" <notifications@github.com>:
